react/wbooks/src/app/components/SignUp/index.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import { t } from 'i18next';
+
+import imagePlaceholder from '../../assets/wolox-logo.png';
+import InputLabel from '../InputLabel';
+
+import styles from './styles.module.scss';
+
+class SignUp extends Component {
+  handleSignUp = () => {
+    console.log({
+      user: {
+        email: this.state.email,
+        password: this.state.password,
+        /* eslint-disable camelcase */
+        password_confirmation: this.state.passwordConfirmation,
+        first_name: this.state.name,
+        last_name: this.state.lastname,
+        /* eslint-enable camelcase */
+        locale: 'en'
+      }
+    });
+  };
+
+  onChangeName = name => {
+    this.setState({ name });
+  };
+
+  onChangeLastname = lastname => {
+    this.setState({ lastname });
+  };
+
+  onChangePassword = password => {
+    this.setState({ password });
+  };
+
+  onChangeEmail = email => {
+    this.setState({ email });
+  };
+
+  onChangePasswordConfirmation = passwordConfirmation => {
+    this.setState({ passwordConfirmation });
+  };
+
+  render() {
+    return (
+      <div className={`${styles.container} column background-wild-sand`}>
+        <img src={imagePlaceholder} alt="Wolox logo" className={styles.woloxLogoImage} />
+        <form onSubmit={this.handleSignUp} className={`${styles.signUpForm} m-bottom-3`}>
+          <InputLabel
+            className=""
+            textClassName={`${styles.inputLabel} m-top-3`}
+            dataFor="name"
+            label={t('SignUp:name')}
+            inputClassName={`${styles.input} full-width`}
+            name="name"
+            inputId="name"
+            inputType="text"
+            handleChange={this.onChangeName}
```

Comment:
If the input label sends you the value and the name, you can use the same function for all the inputs in this form.

`onValueChange = (name, value) => this.setState({ [name]: value })`

or you can send the evt.target and do something like that:

`onValueChange = ({ id, value }) => this.setState({ [id]: value })`

And all the inputs are very similar, only change some props. So you can create an array with the props of the inputs and then use a map for render it. For example:

`FIELDS.map(props =>
<InputLabel
textClassName={'${styles.inputLabel} m-top-3'}
label={props.label}
.../>)`

Reply:
awesome :)